cmake/common/targets.cmake

```diff
@@ -56,7 +56,12 @@ function(silkworm_library TARGET)
   list(FILTER TEST_SRC INCLUDE REGEX "${TEST_REGEX}")
 
   list(FILTER SRC EXCLUDE REGEX "${TEST_REGEX}")
+  # cli subdirectories with CMakeLists.txt belong to silkworm_*_cli libraries
+  if(NOT "${CMAKE_CURRENT_SOURCE_DIR}" MATCHES "/cli$")
+    list(FILTER SRC EXCLUDE REGEX "\/cli\/")
+  endif()
```

Comment:
This should be placed right after `list_filter_dirs(SRC SUB_LIBS)` line, otherwise TEST_SRC will get "cli/*" sources included. Perhaps we should rename the initial SRC to ALL_SRC, and then `set(SRC ${ALL_SRC})` to avoid this confusion. Sorry for this.

Reply:
I've moved it.